### PR TITLE
Cherry-picks the job refactor + traitor changes from upstream to master ahead of time

### PR DESCRIPTION
**Pull request by MrMelbert**
- Cherrypicks the /tg/ job refactor ahead of time and fixes / adapts all the problems to our modular content.
- Cherrypicks the /tg/ traitor changes to work around any issues with advanced traitors
- Adjusts infiltrator slightly.
- Repaths advanced traitors
- Refactors unique-examine element almost entirely.

### CODE REVIEW

**Comment by Jolly-66** [comment]
>233 files
oh dear god

File: jollystation_modules/code/modules/uplink/uplink_infiltrator.dm [hunk 1]
```diff
@@ -51,7 +51,7 @@
 	purchasable_from = UPLINK_NUKE_OPS | UPLINK_CLOWN_OPS | UPLINK_INFILTRATOR
 
 /datum/uplink_item/stealthy_weapons/cqc/infiltrator // Do I really want to give out CQC for non-ops?
-	cost = 24
+	cost = 30 // ...Yes, but it costs their entire budget
```
Comment (Jolly-66) [inline jollystation_modules/code/modules/uplink/uplink_infiltrator.dm]: wrong branch?

Comment (MrMelbert) [reply jollystation_modules/code/modules/uplink/uplink_infiltrator.dm]: you reviewed an outdated commit

**Comment by MrMelbert** [comment]
advanced traitor panel definitely gonna need some love after this is done

**Pull request merged**